frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx

```diff
@@ -172,22 +172,26 @@ export default class ExpressionEditorTextfield extends React.Component {
     const suggestion = suggestions && suggestions[index];
 
     if (suggestion) {
-      let prefix = source.slice(0, suggestion.index);
-      if (suggestion.prefixTrim) {
-        prefix = prefix.replace(suggestion.prefixTrim, "");
-      }
-      let postfix = source.slice(suggestion.index);
-      if (suggestion.postfixTrim) {
-        postfix = postfix.replace(suggestion.postfixTrim, "");
-      }
-      if (!postfix && suggestion.postfixText) {
-        postfix = suggestion.postfixText;
+      const { tokens } = tokenize(source);
+      const token = tokens.find(t => t.end >= suggestion.index);
+      if (token) {
+        const prefix = source.slice(0, token.start);
+        const postfix = source.slice(token.end);
+        const suggested = suggestion.text;
+        const paren = _.last(suggested) === "(";
```

Comment:
is there shenanigans with escaped paren? should we care about any possible shenanigans with escaped paren?

Reply:
I can't think of any.
This `(` is only padded for function names and eventually, once I fix all the callee sites, it can go away as well.

Reply:
yeah, I can see the path to it going away, nicely enough

Reply:
@flamber found an issue with this logic (see below) so I've tweaked it since.